### ISSUE TITLE
[Bug]: Test coverage report broken for @storybook/nextjs

### ISSUE THREAD

**Issue by rotous** [storybookjs/storybook#26344]
### Describe the bug

- Create a standard NextJS project with the following command: `npx create-next-app@latest`
- Add Storybook with the following command: `npx sb init --builder webpack5`
- Add @storybook/nextjs with the following command: `npm install --save-dev @storybook/nextjs@next`
- Update the config in `.storybook/main.js` with the following line: `framework: '@storybook/nextjs',` (as documented on https://storybook.js.org/docs/8.0/get-started/nextjs)
- Add the test-runner with: `npm install @storybook/test-runner --save-dev`
- Add `"test-storybook": "test-storybook"` to the scripts in package.json
- Add the coverage addon with `npm install @storybook/addon-coverage --save-dev`
- Start Storybook with `npm run storybook`
- Create the coverage report with `npm run test-storybook -- --coverage`
- Output the report to the console with `npx nyc report -t coverage/storybook/`

Result:

 PASS   browser: chromium  src/stories/Header.stories.js
 PASS   browser: chromium  src/stories/Button.stories.js
 PASS   browser: chromium  src/stories/Page.stories.js

Test Suites: 3 passed, 3 total
Tests:       8 passed, 8 total
Snapshots:   0 total
Time:        1.711 s, estimated 2 s
Ran all test suites.
Coverage file (2 bytes) written to .nyc_output/coverage.json

File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |

### System

```bash
Storybook Environment Info:

  System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (20) x64 13th Gen Intel(R) Core(TM) i7-13700H
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v20.11.0/bin/npm <----- active
  npmPackages:
    @storybook/addon-coverage: ^1.0.1 => 1.0.1
    @storybook/addon-essentials: ^7.6.17 => 7.6.17
    @storybook/addon-interactions: ^7.6.17 => 7.6.17
    @storybook/addon-links: ^7.6.17 => 7.6.17
    @storybook/addon-onboarding: ^1.0.11 => 1.0.11
    @storybook/blocks: ^7.6.17 => 7.6.17
    @storybook/nextjs: ^7.6.17 => 7.6.17
    @storybook/react: ^7.6.17 => 7.6.17
    @storybook/test: ^7.6.17 => 7.6.17
    @storybook/test-runner: ^0.16.0 => 0.16.0
    eslint-plugin-storybook: ^0.8.0 => 0.8.0
    storybook: ^7.6.17 => 7.6.17
```

### Additional context

The issue seems to be introduced with `@storybook/nextjs@7.6.0-alpha.5`. The coverage reporting works fine up until `@storybook/nextjs@7.6.0-alpha.4`:

````
Storybook Environment Info:

  System:
    OS: Linux 5.15 Ubuntu 22.04.3 LTS 22.04.3 LTS (Jammy Jellyfish)
    CPU: (20) x64 13th Gen Intel(R) Core(TM) i7-13700H
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 20.11.0 - ~/.nvm/versions/node/v20.11.0/bin/node
    npm: 10.5.0 - ~/.nvm/versions/node/v20.11.0/bin/npm <----- active
  npmPackages:
    @storybook/addon-coverage: ^1.0.1 => 1.0.1
    @storybook/addon-essentials: ^7.6.17 => 7.6.17
    @storybook/addon-interactions: ^7.6.17 => 7.6.17
    @storybook/addon-links: ^7.6.17 => 7.6.17
    @storybook/addon-onboarding: ^1.0.11 => 1.0.11
    @storybook/blocks: ^7.6.17 => 7.6.17
    @storybook/nextjs: ^7.6.0-alpha.4 => 7.6.0-alpha.4
    @storybook/react: ^7.6.17 => 7.6.17
    @storybook/test: ^7.6.17 => 7.6.17
    @storybook/test-runner: ^0.16.0 => 0.16.0
    eslint-plugin-storybook: ^0.8.0 => 0.8.0
    storybook: ^7.6.17 => 7.6.17
````
Output of `$ npx nyc report -t coverage/storybook/`:

File        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
------------|---------|----------|---------|---------|-------------------
All files   |   93.93 |    77.77 |   66.66 |    87.5 |
 Button.jsx |     100 |    71.42 |     100 |     100 | 19
 Header.jsx |     100 |      100 |     100 |     100 |
 Page.jsx   |   81.81 |      100 |      50 |   66.66 | 14-15

**Comment by yamish123** [comment]
I have same issue too.

**Comment by valentinpalkovic** [comment]
Hi @rotous 

This got very likely fixed in Storybook 8.0.6. Could you please try it out?

**Comment by yamish123** [comment]
@valentinpalkovic 
Thanks! This issue is solved for me at Storybook 8.0.6.

**Comment by rotous** [comment]
Finally had time to update. The issue is fixed for me too with Storybook v8.0.8. Thanks!
